internal/api/handlers/upsert_test.go

```diff
@@ -0,0 +1,104 @@
+package handlers
+
+import (
+	"awans.org/aft/internal/api"
+	"awans.org/aft/internal/bus"
+	"awans.org/aft/internal/db"
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestUpsertUpdate(t *testing.T) {
+	appDB := db.NewTest()
+	eventbus := bus.New()
+	db.AddSampleModels(appDB)
+
+	tx := appDB.NewRWTx()
+	jsonString := `{ "firstName":"Andrew", "lastName":"Wansley", "age": 32, "emailAddress":"andrew.wansley@gmail.com"}`
+	u := api.MakeRecord(appDB.NewTx(), "user", jsonString)
+	tx.Insert(u)
+	tx.Commit()
+
+	req, err := http.NewRequest("POST", "/user.upsert", strings.NewReader(
+		`{"update":{
```

Comment:
this request should fail as invalid -- upsert needs to require create *and* update